WordPressVIPMinimum/Sniffs/Functions/DynamicCallsSniff.php

```diff
@@ -0,0 +1,271 @@
+<?php
+/**
+ * WordPressVIPMinimum Coding Standard.
+ *
+ * @package VIPCS\WordPressVIPMinimum
+ */
+
+namespace WordPressVIPMinimum\Sniffs\Functions;
+
+use PHP_CodeSniffer_File as File;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * This sniff enforces that certain functions are not
+ * dynamically called.
+ *
+ * An example:
+ *
+ * <code>
+ *   $func = 'func_num_args';
+ *   $func();
+ * </code>
+ *
+ * See here: http://php.net/manual/en/migration71.incompatible.php
+ *
+ * Note that this sniff does not catch all possible forms of dynamic
+ * calling, only some.
+ */
+class DynamicCallsSniff implements \PHP_CodeSniffer_Sniff {
+	/**
+	 * Functions that should not be called dynamically.
+	 *
+	 * @var array
+	 */
+	private $_blacklisted_functions = array(
+		'assert',
+		'compact',
+		'extract',
+		'func_get_args',
+		'func_get_arg',
+		'func_num_args',
+		'get_defined_vars',
+		'mb_parse_str',
+		'parse_str',
+	);
+
+	/**
+	 * Array of functions encountered, along with their values.
+	 * Populated on run-time.
+	 *
+	 * @var array
+	 */
+	private $_variables_arr = array();
+
+	/**
+	 * Returns the token types that this sniff is interested in.
+	 *
+	 * We want everything variable- and function-related.
+	 *
+	 * @return array(int)
+	 */
+	public function register() {
+		return array( T_VARIABLE => T_VARIABLE ) +
+			Tokens::$functionNameTokens;
```

Comment:
It does not look like the script is starting of function name tokens. Only variables are being tested. Let's stick with T_VARIABLE in here only.